### PR TITLE
feat(NcHeaderButton): Add a button only alternative for the header menu

### PR DESCRIPTION
**Pull request by susnux**
### ☑️ Resolves

This allows us to have one single place for styling the header actions, as currently the button-only variants use custom styles (unified search, assistent).

### 🖼️ Screenshots

Screenshot 2024-09-03 at 15-43-49 Nextcloud Vue Style Guide

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [x] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

**Review by marcoambrosini — APPROVED** [review]

File: src/components/NcHeaderMenu/header-menu__trigger.scss [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	&--opened &__trigger,
+	&__trigger:hover,
+	&__trigger:focus,
+	&__trigger:active {
+		opacity: 1;
+	}
+}
+
+@media only screen and (max-width: $breakpoint-small-mobile) {
```
Comment (Antreesy) [inline src/components/NcHeaderMenu/header-menu__trigger.scss:29]: NcButton is still considered `large`, and  might cause horizontal scrolling on small viewports (see Contacts Menu).

Can we apply something like this?

[fix_NcHeader.patch](https://github.com/user-attachments/files/16905648/fix_NcHeader.patch)

**Review by Antreesy — APPROVED** [review]
Nice addition!

File: src/components/NcHeaderMenu/header-menu__trigger.scss (hunk 1 above)
Comment (susnux) [reply src/components/NcHeaderMenu/header-menu__trigger.scss:29]: > NcButton is still considered `large`

Yes because I did not yet rebase after #6047

Comment (Antreesy) [reply src/components/NcHeaderMenu/header-menu__trigger.scss:29]: Aye aye

**Pull request merged**